snap/snapshots.go

```diff
@@ -27,29 +27,79 @@ import (
 	"path/filepath"
 	"strings"
 
-	"gopkg.in/yaml.v2"
-
 	"github.com/snapcore/snapd/strutil"
+	"gopkg.in/yaml.v2"
 )
 
 var osOpen = os.Open
 
-// SnapshotOptions contains the information provided by a snap package about
-// what files should be included in the snapshots.
+// SnapshotOptions describes the options available for snapshots.
+// The initial source of these options is a file in the snap package.
+// In addition, options can be modified with dynamic requests via REST API.
 type SnapshotOptions struct {
-	// ExcludePaths is the list of file and directory patterns that need to be
+	// Exclude is the list of file and directory patterns that need to be
 	// excluded from a snapshot. At the moment the only supported globbing
 	// character is "*", which stands for any sequence of characters other than
 	// "/".
-	ExcludePaths []string `yaml:"exclude"`
+	Exclude []string `yaml:"exclude" json:"exclude,omitempty"`
 }
 
 const (
 	snapshotManifestPath = "meta/snapshots.yaml"
 )
 
+// Unset determines if the SnapshotOptions object contains values worth
+// marshalling to metadata (not just keys, braces and brackets)
```

Comment:
The `(not just keys, braces and brackets)` part is a bit incongruous with the actual check

Reply:
I think it means `{"exclude:":[]}` but is not the clearest way to express that.

Reply:
Improved.